### PR TITLE
Checks permission for existing projects.

### PR DESCRIPTION
**Pull request by enriquejosepadilla**
Fixes: https://github.com/GoogleCloudPlatform/django-cloud-deploy/issues/205

User study would provide users with projects they are not owner of GAE project, or they were not editors for GKE. This checks to make sure and crashes early.

Only applies to existing-projects.

### CODE REVIEW

**Comment by wangxf123456** [comment]
If the integration tests fail, you can merge with the master branch and try again.

**Comment by enriquejosepadilla** [comment]
the change from copy.deepcopy to dict(args) was due to a very wierd error arising when trying to run locally. As @brianquinlan had suggested using this approach (downside is that it is shallow), i think for future debugging is better than future possibility of nested args.

File: django_cloud_deploy/cli/prompt.py [hunk 1]
```diff
@@ -541,50 +543,102 @@ class GoogleExistingProjectId(TemplatePrompt):
...
+
+        owners = []
+        editors = []
+        if owner_permission:
+            owners = owner_permission[0].get('members', [])
+        if editor_permission:
+            editors = editor_permission[0].get('members', [])
+
+        active_account = re.escape(active_account)
+        if re.search(active_account, ' '.join(owners)):
```
Comment (wangxf123456) [inline django_cloud_deploy/cli/prompt.py]: For example, my account is "tim@gmail.com", and this account does not exist in either owners or editors. And there is another account called "blablatim@gmail.com", who is the owner. Then re.search will return true when active_account is "tim@gmail.com", but it is not necessarily an owner.

File: django_cloud_deploy/cli/prompt.py [hunk 2]
```diff
@@ -1160,9 +1215,12 @@ def prompt(self, command: Command, console: io.IO,
         step_template = '<b>[{}/{}]</b>'
         first_step = step_template.format(1, total_steps)
 
-        creds = self._get_creds(console, first_step, args)
+        auth_client = auth.AuthClient()
+        creds = self._get_creds(console, first_step, args, auth_client)
+        active_account = auth_client.get_active_account()
```
Comment (wangxf123456) [inline django_cloud_deploy/cli/prompt.py]: auth_client.get_active_account() will return an empty string or None (I am not sure whether it is None or empty string in python) if the user has not logged in with gcloud. Will this be a problem?

File: django_cloud_deploy/tests/unit/cli/prompt_test.py [hunk 3]
```diff
@@ -286,6 +303,25 @@ def test_existing_prompt(self, unused_mock):
...
+        test_io = io.TestIO()
+        args = {
+            'backend': 'gae',
+            'project_creation_mode': workflow.ProjectCreationMode.MUST_EXIST,
+            'project_id': 'projectid-123',
+            'use_existing_project': True
+        }
+        test_io.answers.append('projectid-123')
+
+        with self.assertRaises(SystemExit):
```
Comment (wangxf123456) [inline django_cloud_deploy/tests/unit/cli/prompt_test.py]: Why SystemExit?

File: django_cloud_deploy/cli/prompt.py (hunk 2 above)
Comment (enriquejosepadilla) [reply django_cloud_deploy/cli/prompt.py]: it returns an empty string I confirmed this manually. but I can add a check to make sure it doesnt crash.

File: django_cloud_deploy/cli/prompt.py (hunk 1 above)
Comment (enriquejosepadilla) [reply django_cloud_deploy/cli/prompt.py]: that is a good catch.

File: django_cloud_deploy/tests/unit/cli/prompt_test.py (hunk 3 above)
Comment (enriquejosepadilla) [reply django_cloud_deploy/tests/unit/cli/prompt_test.py]: if the user passes it as a flag it does a system exit. I forgot to add a test so it just pops out a new answer in the append.

File: django_cloud_deploy/cli/prompt.py (hunk 2 above)
Comment (enriquejosepadilla) [reply django_cloud_deploy/cli/prompt.py]: There was a check already asserting it is not an empty string, as the user must have been logged in by that point.

**Pull request merged**